### PR TITLE
BYO Uploader

### PR DESCRIPTION
**Pull request by timkelty**
### Description
Makes it possible to register JS uploaders, allowing uploads to bypass PHP and stream directly from the browser.

For posterity, here's all the places that should be tested when fussing with uploaders:

- [x] Asset Detail
  - [x] Replace file button
- [x] Asset Index
  - [x] Replace file element action
  - [x] Upload files button (with multiple files)
  - [x] Upload files button (with filename conflicts)
  - [x] Move asset
- [x] Entry Detail
  - [x] Upload files button
  - [x] Upload files button (with selection condition)
- [x] Rebrand upload
- [x] Member photo

### Related issues
- https://github.com/craftcms/cms/discussions/9084

### CODE REVIEW

File: src/web/assets/cp/src/js/AssetSelectInput.js [hunk 1]
```diff
@@ -260,54 +263,54 @@ Craft.AssetSelectInput = Craft.BaseElementSelectInput.extend({
...
+    var progress = parseInt(Math.min(data.loaded / data.total, 1) * 100, 10);
     this.progressBar.setProgressPercentage(progress);
   },
 
   /**
    * On a file being uploaded.
    */
   _onUploadComplete: function (event, data) {
-    if (data.result.error) {
-      Craft.cp.displayError(data.result.error);
```
Comment (timkelty) [inline src/web/assets/cp/src/js/AssetSelectInput.js:272]: Fixing stray legacy error response handling

File: src/web/assets/cp/CpAsset.php [hunk 2]
```diff
@@ -476,6 +476,7 @@ private function _craftData(): array
             'isMultiSite' => Craft::$app->getIsMultiSite(),
             'limitAutoSlugsToAscii' => $generalConfig->limitAutoSlugsToAscii,
             'maxUploadSize' => Assets::getMaxUploadSize(),
+            'maxUploadFileSize' => Craft::$app->getConfig()->getGeneral()->maxUploadFileSize,
```
[the remark this answers is not in the document]
Comment (timkelty) [reply src/web/assets/cp/CpAsset.php:480]: @timkelty might need to normalize `maxUploadFileSize`

File: src/elements/Asset.php [hunk 3]
```diff
@@ -1493,33 +1494,54 @@ public function getAdditionalButtons(): string
...
+                let {message, filename} = response || {};
+                
+                if (!message) {
+                    message = filename
+                        ? Craft.t('app', 'Replace file failed for “{filename}”.', {filename})
+                        : Craft.t('app', 'Replace file failed.');
+                }
+                
+              Craft.cp.displayError(message);
+            },
```
Comment (timkelty) [inline src/elements/Asset.php:1541]: Replaced the vanilla js `alert` with `Craft.cp.displayError` while I was in there

File: src/templates/users/_photo.twig [hunk 4]
```diff
@@ -4,7 +4,7 @@
         <div id="current-photo">
             {% set photoUrl = user.getThumbUrl(100) %}
             {% if photoUrl %}
-                <img sizes="100px" srcset="{{ photoUrl }} 100w, {{ user.getphotoUrl(200) }} 200w" alt="{{ user.getThumbAlt() }}">
+                <img sizes="100px" srcset="{{ photoUrl }} 100w, {{ user.getThumbUrl(200) }} 200w" alt="{{ user.getThumbAlt() }}">
```
Comment (timkelty) [inline src/templates/users/_photo.twig:7]: Fixing bug from #13262

**Review by brandonkelly — APPROVED** [review]

**Pull request merged**